src/backend/catalog/aclchk.c

```diff
@@ -1722,6 +1723,11 @@ ExecGrant_Attribute(InternalGrant *istmt, Oid relOid, const char *relname,
 
 	pfree(merged_acl);
 
+	if (bbf_execute_grantstmt_as_dbsecadmin_hook)
```

Comment:
Can we skip this if grantor is sysadmin?

Reply:
Or may be skip if we already have all the required permissions from the current best grantor.

Reply:
Incorporated this changes in hooks implementation.